kubernetes-itests/src/test/java/io/fabric8/kubernetes/CreateOrReplaceIT.java

```diff
@@ -232,6 +239,98 @@ public void testCreateOrReplaceIngress() {
     client.network().ingresses().inNamespace(session.getNamespace()).withName(ingress.getMetadata().getName()).delete();
   }
 
+  @Test
+  public void testCreateOrReplaceGenericResource() {
+    // Given
+    ConfigMap configMap = new ConfigMapBuilder()
+      .withNewMetadata().withName("resource-cm-1").endMetadata()
+      .addToData("a1", "A1")
+      .addToData("a2", "A2")
+      .build();
+
+    // When
+    ConfigMap createdResource = client.resource(configMap).inNamespace(session.getNamespace()).createOrReplace();
+    configMap.setData(Collections.singletonMap("b1", "B1"));
+    ConfigMap replacedResource = client.resource(configMap).inNamespace(session.getNamespace()).createOrReplace();
+
+    // Then
+    assertNotNull(createdResource);
+    assertEquals(2, createdResource.getData().size());
+    assertEquals("A1", createdResource.getData().get("a1"));
+    assertEquals("A2", createdResource.getData().get("a2"));
+    assertNotNull(replacedResource);
+    assertEquals(1, replacedResource.getData().size());
+    assertEquals("B1", replacedResource.getData().get("b1"));
+    // Cleanup
+    client.resource(configMap).inNamespace(session.getNamespace()).delete();
```

Comment:
this belongs to a teardown method.
If this test fails, it won't be cleaned up and probably the rest of tests that run after will fail too (if this cleanup is really necessary)